packages/message/tests/test-StepFunctions.js

```diff
@@ -2,9 +2,472 @@
+});
+
+test('getCumulusMessageFromExecutionEvent() returns the event input for a RUNNING event', async (t) => {
+  const event = {
+    detail: {
+      status: 'RUNNING',
+      input: JSON.stringify({
+        cumulus_meta: {
+          workflow_start_time: 122,
+        },
+      }),
+      startDate: 123,
+      stopDate: null,
+    },
+  };
+
+  const message = await getCumulusMessageFromExecutionEvent(event);
+
+  const expectedMessage = {
+    cumulus_meta: {
+      workflow_start_time: 122,
+      workflow_stop_time: null,
+    },
+    meta: {
+      status: 'running',
+    },
+  };
+
+  t.deepEqual(message, expectedMessage);
+});
+
+test('getCumulusMessageFromExecutionEvent() returns the event output for a SUCCEEDED event', async (t) => {
+  const event = {
+    detail: {
+      status: 'SUCCEEDED',
+      output: JSON.stringify({
+        cumulus_meta: {
+          workflow_start_time: 122,
+        },
+      }),
+      startDate: 123,
+      stopDate: 124,
+    },
+  };
+
+  const message = await getCumulusMessageFromExecutionEvent(event);
+
+  const expectedMessage = {
+    cumulus_meta: {
+      workflow_start_time: 122,
+      workflow_stop_time: 124,
+    },
+    meta: {
+      status: 'completed',
+    },
+  };
+
+  t.deepEqual(message, expectedMessage);
+});
+
+test.serial('getCumulusMessageFromExecutionEvent() returns the failed execution message for a failed event', async (t) => {
+  const input = {
+    cumulus_meta: {
+      state_machine: 'arn:aws:states:us-east-1:111122223333:stateMachine:HelloWorld-StateMachine',
+      execution_name: 'my-execution-name',
+      workflow_start_time: 122,
+    },
+  };
+
+  const event = {
+    detail: {
+      status: 'FAILED',
+      input: JSON.stringify(input),
+      startDate: 123,
+      stopDate: 124,
+    },
+  };
+
+  const failedTaskOutput = input;
+
+  const getExecutionHistoryFunction = ({ executionArn }) => {
+    if (executionArn !== 'arn:aws:states:us-east-1:111122223333:execution:HelloWorld-StateMachine:my-execution-name') {
+      throw new Error(`Expected executionArn === 'arn:aws:states:us-east-1:111122223333:execution:HelloWorld-StateMachine:my-execution-name' but got ${executionArn}`);
+    }
+    return {
+      events: [
+        {
+          // lastStepFailedEvent
+          type: 'LambdaFunctionFailed',
+          id: 3,
+        },
+        {
+          // failedStepExitedEvent
+          type: 'TaskStateExited',
+          previousEventId: 3,
+          stateExitedEventDetails: {
+            output: JSON.stringify(failedTaskOutput),
+          },
+        },
+      ],
+    };
+  };
+
+  const message = await getCumulusMessageFromExecutionEvent(event, getExecutionHistoryFunction);
+
+  const expectedMessage = {
+    cumulus_meta: {
+      state_machine: 'arn:aws:states:us-east-1:111122223333:stateMachine:HelloWorld-StateMachine',
+      execution_name: 'my-execution-name',
+      workflow_start_time: 122,
+      workflow_stop_time: 124,
```

Comment:
```suggestion
      workflow_stop_time: event.detail.stopDate,
```

Not necessary per se